### PR TITLE
Add MddGetPackageGraphRevisionId() + M.W.AM.DynamicDependency.PackageDependency.PackageGraphRevisionId API Spec

### PR DESCRIPTION
**Pull request by DrusTheAxe**
Add `MddGetPackageGraphRevisionId()` (Flat-C API) and `PackageGraphRevisionId` property (WinRT API), for alignment with [Windows 11 API / terminology](https://docs.microsoft.com/en-us/windows/apps/desktop/modernize/framework-packages/use-the-dynamic-dependency-api).

GenerationId is soft-deprecated (predecessor equivalent to PackageGraphRevisionId still supported but will go away in a future release).

Also corrected spec missing updates:
* Corrected missing changes per previous API Review and implementation
* Added missing contracts (added in WinAppSDK 1.1)

https://task.ms/41077950

### CODE REVIEW

**Comment by DrusTheAxe** [comment]
We're doing an "email review" of this - please leave your comments in the PR by 5pm Seattle time, 2022-08-26. @DrusTheAxe will address feedback in future updates.

**Review by EHO-makai — COMMENTED** [review]
Is it on purpose that we change generation => revision id but make no mention of the "soft deprecation?"

**Comment by DrusTheAxe** [comment]
> Is it on purpose that we change generation => revision id but make no mention of the "soft deprecation?"

Are you referring to a specific line?

**Comment by EHO-makai** [comment]
No, it seems like most of the edits just change generation => revision.  But no mention is made of what you're supposed to do if you had previously used generation.  I'm not asking for a change here necessarily, just wondering if silently doing a "replace" here is what you intended / OK.

**Comment by DrusTheAxe** [comment]
> No, it seems like most of the edits just change generation => revision. But no mention is made of what you're supposed to do if you had previously used generation.

2 words for the same thing.

Was originally called GenerationId in Windows (internal) and then WinAppSDK DynDep API Review. Windows had no public API to get that value so we fixed that with a new public API (`GetPackageGraphRevisionId`) per API Review, which recommended using the term 'revision id' instead of 'generation id'.

This merely updates WinAppSDK to support consist lingo with Windows.

>I'm not asking for a change here necessarily, just wondering if silently doing a "replace" here is what you intended / OK.

Yes. Given it's the same thing, the GetGeneration doc-comments point you to GetRevision, and it's open source so anyone who looks can see it's the same thing<sup>1</sup> I didn't see a need to belabor the point. I expect the release notes and docs will mention this.


<sup>1</sup> The implementation is literally 1-liners `UINT32 MddGetGenerationId() { return MddGetPackageGraphRevisionId(); }`)

**Review by EHO-makai — APPROVED** [review]

**Comment by MikeHillberg** [comment]
We have a [template for API specs](https://github.com/microsoft/WindowsAppSDK/blob/main/specs/spec_template.md) (not just for WinRT APIs)

File: specs/dynamicdependencies/DynamicDependencies.md [hunk 1]
```diff
@@ -1841,12 +1850,19 @@ runtimeclass PackageDependency
...
+    /// Successful calls change the package graph's current revision id.
     ///
-    /// @see GenerationId
+    /// @see PackageGraphRevisionId
     PackageDependencyContext Add(AddPackageDependencyOptions options);
 
-    /// Return the package graph's current generation id.
+    /// Return the package graph's current revision id.
+    [contract(DynamicDependencyContract, 2)]
+    static UInt32 PackageGraphRevisionId{ get; };
```
Comment (MikeHillberg) [inline specs/dynamicdependencies/DynamicDependencies.md:1860]: Standard guideline is to use Int32, even if signed is invalid, unless the API really needs more than 2B values

Comment (DrusTheAxe) [reply specs/dynamicdependencies/DynamicDependencies.md:1860]: That's news. This is a wrapper over the Win32 API which returns `uint32`. Still worth going `int32`?

Comment (idigdoug) [reply specs/dynamicdependencies/DynamicDependencies.md:1860]: Existing functions already return U32, so probably better to just continue with that.

**Pull request merged**